salt/state.py

```diff
@@ -2326,8 +2326,7 @@ def check_requisite(self, low, running, chunks, pre=False):
                     continue
                 if run_dict[tag].get('proc'):
                     # Run in parallel, first wait for a touch and then recheck
-                    time.sleep(0.01)
-                    return self.check_requisite(low, running, chunks, pre)
+                    run_dict[tag].get('proc').join()
```

Comment:
Eeeeeek. Will this block?

Reply:
This will create a zombie until parent quits.

Reply:
@cachedout yes
@isbm according to  https://docs.python.org/2/library/multiprocessing.html#all-platforms join will waitpid and reap, unless im misreading

Reply:
Yes. But `.join()` will block everything. And if you did not set it daemonic, it will anyway be joined after the start.